From fd8137f281f83312c1964f70166286d6fc255823 Mon Sep 17 00:00:00 2001
From: Harish Rachakonda <quic_rachakon@quicinc.com>
Date: Mon, 19 Aug 2024 09:29:38 +0530
Subject: [PATCH] hostapd: Disabling ht_enabled for 6G frequencies

Currently during channel switch ht_enabled is enabled for 6G because of
which hostapd_cli -i wlanX status is showing wrong values

Fix this issue by disabling ht_enabled for 6G frequencies.

Patch-dependency: None
Patch-work: None

Signed-off-by: Harish Rachakonda <quic_rachakon@quicinc.com>
---
 src/drivers/driver_nl80211_event.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/src/drivers/driver_nl80211_event.c
+++ b/src/drivers/driver_nl80211_event.c
@@ -1269,6 +1269,9 @@ static void mlme_event_ch_switch(struct
 		wpa_printf(MSG_WARNING, "nl80211: Unknown secondary channel information - following channel definition calculations may fail");
 	}
 
+	if (is_6ghz_freq(nla_get_u32(freq)))
+		ht_enabled = 0;
+
 	os_memset(&data, 0, sizeof(data));
 	data.ch_switch.freq = nla_get_u32(freq);
 	data.ch_switch.ht_enabled = ht_enabled;
